src/core/ports/commands/get-cycle-delta.command.ts

```diff
@@ -16,12 +16,93 @@ export class GetCycleDeltaCommand extends BaseDeltaCommand {
 
   public async execute(CycleID: string): Promise<Delta> {
     const comparisonDate = this.getComparisonDate()
+    const row = await this.core.entityManager.query(
+      `
+        with latest_check_in as (
+          select distinct on (krci.key_result_id) * from public.key_result_check_in krci
+          order by krci.key_result_id, krci.created_at desc
+        ),
+        latest_check_in_week_before as (
```

Comment:
Seguindo essa mesma lógica vale a pena extrair alguns CTEs pra views pra vc poder reaproveitar nos outros commands e centralizar algumas regras de negócio

Reply:
Vai ser bem útil fazer iss pra classificação de key results em status/confidence, por exemplo

Reply:
@danilo-valente você sabe alguma maneira de "passar um parâmetro" para views? 
Tipo, hoje fazemos o "delta" de progresso da semana passada para a atual, existe um mundo que essa frequência mude para diária (considerando uma mudança de posicionamento para team_management).

Seria lindo isso, no nosso caso é o que passamos como 'comparisonDate' em alguma queries. 

Mas pelo que eu vi isso só é possível em funções, aí talvez seja melhor deixar o cálculo em cada arquivo sem views.

Reply:
Hm, tem várias maneiras que daria para fazer isso, mas eu acho que o mais fácil hoje é ter uma view pra cada (weekly e daily) ou até uma única view com 2 colunas (week_delta e day_delta). Dá uma lida [nesse SO](https://dba.stackexchange.com/questions/151169/are-views-harmful-for-performance-in-postgresql) só pra confirmar que não vai ter nenhum impacto em performance mesmo.
Outra alternativa é usar stored functions com parâmetros ao invés de views para casos como esses.

Aproveitando, acho que vale a pena mudar a forma que as comparações de data são usadas. Ao invés de calcular o `comparisonDate` na aplicação fica mais fácil parametrizar passando um inteiro e usar pra calcular a diferença já na própria query (em dias, por exemplo):

```sql
where created_at >= now() - interval '7' day
```

Reply:
Boa, sobre os 2 pontos:
1.  Se views atrapalham performance, a resposta do [Andriy M](https://dba.stackexchange.com/users/6965/andriy-m) bate com o que eu conhecia, vou fazer uns testes com explain analyse para confirmar, mas só tive que alterar 1 coluna para "casar" todas as views, e a alteração é puxar mais uma coluna, não teve nenhum processamento a mais

2. Vou só confirmar se podemos utilizar essa regra, pq pelo que entendo, o `comparisonDate` é comparando com o último sábado, aí estaríamos alterando regras de negócio. Mas pra mim, faz sentido

Reply:
1. Pelo que eu entendi as views não atrapalham a performance por si só, porque elas funcionam exatamente como uma query. Então os princípios de otimização de queries (índices, particionamento, etc) se aplicam igual
2. Ainda vale, só vai mudar a função ([referência](https://stackoverflow.com/questions/56036043/how-to-get-last-sunday-including-today-in-postgresql)). Meu ponto é só para também jogar esse cálculo de datas pro banco porque pode facilitar a escrita de algumas queries

Reply:
Tudo feito, para via de comparação

Query mais simples, que **não** utiliza todas as colunas das views (get-team-status):
Query pura:  143 ms
Utilizando views compartilhadas: 249ms

Query mais complexa, que utiliza todas as colunas das views compartilhadas (get-cycle-delta):
Query pura: 289 ms
Utilizando views compartilhadas: 293ms (deve ser igual, só o desvio normal)

Reply:
Dado esse cenário, entendo que seja melhor deixar a query mais "pura", sem tentar criar uma view that solve all problems, vou criar umas essas views mais simples para facilitar a legibilidade

Reply:
Com a query com views simplificadas do get-team-status : 140ms, esperado